contracts/optimistic-ethereum/OVM/bridge/OVM_L1CrossDomainMessenger.sol

```diff
@@ -140,6 +143,7 @@ contract OVM_L1CrossDomainMessenger is iOVM_L1CrossDomainMessenger, OVM_BaseCros
     )
         override
         public
+        nonReentrant
```

Comment:
FWIW, this code path speaks directly to the rollup contracts (`OVM_CanonicalTransactionChain.enqueue` specifically) so it should not be re-enterable based on the behavior of that target, so it should not be needed.

I'd recommend removing it, because the L1 `SSTORE` is gonna be expensive.  If you think that it's worth keeping, we can, but we should also add it to `sendMessage`, since the functions serve similar purposes.

Reply:
`OVM_BaseCrossDomainMessenger.sendMessage()` already has the `nonReentrant` modifier on it. You can see above that I moved it down a couple lines to improve the style. But looking at both `sendMessage` and `replayMessage` more  closely, I agree that it looks pretty safe to remove them.